crates/bevy_anti_aliasing/src/dlss/mod.rs

```diff
@@ -42,52 +44,118 @@ use dlss_wgpu::{
         DlssRayReconstruction, DlssRayReconstructionDepthMode, DlssRayReconstructionRoughnessMode,
     },
     super_resolution::DlssSuperResolution,
+    FeatureSupport,
 };
 use std::{
     marker::PhantomData,
     ops::Deref,
     sync::{Arc, Mutex},
 };
 use tracing::info;
+use uuid::Uuid;
 
-pub use bevy_render::{
-    DlssProjectId, DlssRayReconstructionSupported, DlssSuperResolutionSupported,
-};
-pub use dlss_wgpu::DlssPerfQualityMode;
+/// Initializes DLSS support in the renderer. This must be registered before [`RenderPlugin`](bevy_render::RenderPlugin) because
+/// it configures render init code.
+#[derive(Default)]
+pub struct DlssInitPlugin;
+
+impl Plugin for DlssInitPlugin {
+    fn build(&self, app: &mut App) {
+        let dlss_project_id = app.world().get_resource::<DlssProjectId>()
+                        .expect("The `dlss` feature is enabled, but DlssProjectId was not added to the App before DlssPlugin.").0;
+        let mut raw_vulkan_settings = app
+            .world_mut()
+            .get_resource_or_init::<RawVulkanInitSettings>();
+
+        raw_vulkan_settings.create_instance_callbacks.push(Arc::new(
+            move |mut args, additional_vulkan_features| {
+                let mut feature_support = FeatureSupport::default();
+                match dlss_wgpu::register_instance_extensions(
+                    dlss_project_id,
+                    &mut args,
+                    &mut feature_support,
+                ) {
+                    Ok(_) => {
+                        if feature_support.ray_reconstruction_supported {
+                            additional_vulkan_features.register::<DlssRayReconstructionSupported>();
+                        }
+                        if feature_support.super_resolution_supported {
+                            additional_vulkan_features.register::<DlssSuperResolutionSupported>();
+                        }
+                    }
+                    Err(_) => todo!(),
+                }
+            },
+        ));
+
+        raw_vulkan_settings.create_device_callbacks.push(Arc::new(
+            move |mut args, adapter, additional_vulkan_features| {
+                let mut feature_support = FeatureSupport::default();
+                match dlss_wgpu::register_device_extensions(
+                    dlss_project_id,
+                    &mut args,
+                    adapter,
+                    &mut feature_support,
+                ) {
+                    Ok(_) => {
+                        if feature_support.ray_reconstruction_supported {
+                            additional_vulkan_features.register::<DlssRayReconstructionSupported>();
+                        }
+                        if feature_support.super_resolution_supported {
+                            additional_vulkan_features.register::<DlssSuperResolutionSupported>();
```

Comment:
Theoretically instance creation might support RR/SR, but not device creation. We should technically handle that (which the old code did). Don't think it'll ever occur in practice, though.